src/vdc_collection.rs

```diff
@@ -0,0 +1,205 @@
+
+/// Supported credential formats.
+#[derive(PartialEq)]
+pub enum CredentialFormat {
+    MDoc,
+    W3cJwtVc,
+    W3cJwtVcJsonld,
+    Other(String),  // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String),  // For ease of expansion.
+}
+
+/// An individual credential.  These are mainly for internal use; access to them should go through `VdcCollection`,
+/// below.
+struct Credential {
+    id:      String,           // Probably a UUID, perhaps we should make it one?
+    format:  CredentialFormat,
+    ctype:   CredentialType,
+    tags:    Vec<String>,      // Arbitrary string tags.
+    payload: Vec<u8>,          // The actual credential.
+    dirty:   bool,             // Has this been modified since it was read from storage?
+}
```

Comment:
When working in Rust we almost always use the defaults in Cargo for linting and formatting. Can you run `cargo fmt`? I'd also suggest setting up your text editor to run `cargo fmt` on save.

If you're not already, you should also look into using rust-analyzer, which is the main community LSP. To keep it in sync with your rust installation you should install using `rustup`:  `$ rustup component add rust-analyzer`.

Reply:
Done.